classes/Profile.php

```diff
@@ -226,4 +230,26 @@ private static function findIdTabByAuthSlug($authSlug)
 
         return $result['id_tab'];
     }
+    
+    /**
+     *
+     * @param string $authSlug
+     * @return string
+     */
+    private static function findClassNameByAuthSlug($authSlug)
+    {
+        preg_match(
+            '/ROLE_MOD_[A-Z]+_(?P<classname>[A-Z]+)_(?P<auth>[A-Z]+)/',
+            $authSlug,
+            $matches
+        );
+
+        $result = Db::getInstance()->getRow('
+            SELECT `class_name`
+            FROM `'._DB_PREFIX_.'tab` t
+            WHERE UCASE(`class_name`) = "'.$matches['classname'].'"
+        ');
```

Comment:
This part is duplicated from `function findIdTabByAuthSlug($authSlug)`.
Please write a common function used by `findIdTabByAuthSlug()` and `findClassNameByAuthSlug()`

We should get something like this:

```php

    private static function findByAuthSlug($authSlug)
    {
        preg_match(
            '/ROLE_MOD_[A-Z]+_(?P<classname>[A-Z]+)_(?P<auth>[A-Z]+)/',
            $authSlug,
            $matches
        );
        $result = Db::getInstance()->getRow('
            SELECT `id_tab`, `class_name`
            FROM `'._DB_PREFIX_.'tab` t
            WHERE UCASE(`class_name`) = "'.$matches['classname'].'"
        ');
        return $result;
    }

    /**
     *
     * @param string $authSlug
     * @return int
     */
    private static function findIdTabByAuthSlug($authSlug)
    {
        $tab = $this->findByAuthSlug();
        return $tab['id_tab'];
    }
    
    /**
     *
     * @param string $authSlug
     * @return string
     */
    private static function findClassNameByAuthSlug($authSlug)
    {
        $tab = $this->findByAuthSlug();
        return $tab['class_name'];
    }
```